library/src/lib/calendar/calendar2/calendar2-views/calendar2-month-view/calendar2-month-view.component.ts

```diff
@@ -1,17 +1,104 @@
-import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { Component, Input, Output, ViewEncapsulation, EventEmitter, ElementRef } from '@angular/core';
 
 @Component({
     selector: 'fd-calendar2-month-view',
     templateUrl: './calendar2-month-view.component.html',
     styleUrls: ['./calendar2-month-view.component.scss'],
     encapsulation: ViewEncapsulation.None
 })
-export class Calendar2MonthViewComponent implements OnInit {
+export class Calendar2MonthViewComponent {
+    /** @hidden The id of the calendar passed from the parent component. */
+    @Input() id: string;
 
-    constructor() {
+    /** @hidden A list of month names */
+    @Input() monthNames: string[];
+
+    /** @hidden A number (1-12) representing the selected month */
+    @Input() monthSelected: number;
+
+    /** @hidden When set to true allows escape focus, otherwise resets on beginning. */
+    @Input()
+    allowFocusEscape: boolean = false;
+
+    /** @hidden An event fired when a new month is selected. */
+    @Output()
+    monthClicked: EventEmitter<number> = new EventEmitter<number>();
+
+    constructor(private eRef: ElementRef) {
+    }
+
+    /** Get a number (1-12) representing the current month  */
+    get currentMonth(): number {
+        const tempDate: Date = new Date();
+        return tempDate.getMonth() + 1;
```

Comment:
You can use FdDate.getToday().month to not have to do this

we should avoid repeating work like this and instead have all the (+1) month logic be centralized in FdDate